spec/features/bulkrax_import_spec.rb

```diff
@@ -281,6 +274,153 @@
+              "Authorization" => "Basic dXNlcm5hbWU6cGFzc3dvcmQ=",
+              "Content-Type" => "application/xml;charset=UTF-8",
+              "User-Agent" => "Faraday v0.17.4"
+            }
+          )
+          .to_return(status: 200, body: "", headers: {})
+      end
+
+      context "when the work already has a DOI" do
+        context "the import does not change the DOI status" do
+          let!(:work_1) { create :work, source_identifier: "external-id-1", doi: nil, doi_status_when_public: nil }
+          let!(:work_2) { create :work, source_identifier: "external-id-2", doi: [doi], doi_status_when_public: "draft" }
+          let!(:work_3) { create :work, source_identifier: "external-id-3", doi: [doi], doi_status_when_public: "registered" }
+          let!(:work_4) { create :work, source_identifier: "external-id-4", doi: [doi], doi_status_when_public: "findable" }
+
+          it "does not mint a new DOI" do
+            perform_enqueued_jobs(only: [Bulkrax::ImporterJob, Hyrax::DOI::RegisterDOIJob]) do
+              Bulkrax::ImporterJob.perform_now(importer.id)
+            end
+
+            expect(work_1.reload.doi).to be_empty
+            expect(work_2.reload.doi).to eq([doi])
+            expect(work_3.reload.doi).to eq([doi])
+            expect(work_4.reload.doi).to eq([doi])
+
+            expect(work_1.doi_status_when_public).to be_nil
+            expect(work_2.doi_status_when_public).to eq("draft")
+            expect(work_3.doi_status_when_public).to eq("registered")
+            expect(work_4.doi_status_when_public).to eq("findable")
+          end
+        end
+
+        context "the import changes the DOI status forwards" do
+          let!(:work_1) { create :work, source_identifier: "external-id-1", doi: nil, doi_status_when_public: nil }
+          let!(:work_2) { create :work, source_identifier: "external-id-2", doi: nil, doi_status_when_public: nil }
+          let!(:work_3) { create :work, source_identifier: "external-id-3", doi: [doi], doi_status_when_public: "draft" }
+          let!(:work_4) { create :work, source_identifier: "external-id-4", doi: [doi], doi_status_when_public: "registered" }
+
+          it "changes DOI status and mints only for the correct items" do
+            perform_enqueued_jobs(only: [Bulkrax::ImporterJob, Hyrax::DOI::RegisterDOIJob]) do
+              Bulkrax::ImporterJob.perform_now(importer.id)
+            end
+
+            expect(work_1.reload.doi).to be_empty
+            expect(work_2.reload.doi).to eq([doi])
+            expect(work_3.reload.doi).to eq([doi])
+            expect(work_4.reload.doi).to eq([doi])
+
+            expect(work_1.doi_status_when_public).to be_nil
+            expect(work_2.doi_status_when_public).to eq("draft")
+            expect(work_3.doi_status_when_public).to eq("registered")
+            expect(work_4.doi_status_when_public).to eq("findable")
+          end
+        end
+
+        context "the import changes the DOI status backwards" do
```

Comment:
By taking doi backwards do yo mean to what is  in the imported csv as against the values in work-1 to work-4?